addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1041,6 +1041,122 @@ const ColorpickerUserValueWidget = SelectUserValueWidget.extend({
+
+        return this.containerEl.append(_buildRowElement('', {
+            childNodes: [this.editImageButton, this.removeImageButton],
+        }));
+    },
+    /**
+     * @override
+     */
+    getMethodsParams: function (methodName) {
+        return _.extend({isVideo: this.isVideo}, this._super(...arguments));
+    },
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    updateUI: async function () {
+        await this._super(...arguments);
+        this.removeImageButton.classList.toggle('d-none', !this.isActive());
+        this.editBgTextEl.innerText = this.isActive() ? this._getDisplayName() : this.defaultText;
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Computes the name to display in the edit button.
+     *
+     * @private
+     */
+    _getDisplayName: function () {
+        if (this.isVideo) {
+            return this._value;
+        }
+        return this._value.substring(this._value.lastIndexOf('/') + 1).split(/[?#]/)[0];
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Called when the edit image button is clicked.
+     *
+     * @private
+     */
+    _onEditImage: function (ev) {
+        // Need a dummy element for the media dialog to modify.
+        const dummyEl = document.createElement(this.isVideo ? 'iframe' : 'img');
+        dummyEl.src = this._value;
+        if (this.isVideo) {
+            // Allows the mediaDialog to select the video tab immediately.
+            dummyEl.classList.add('media_iframe_video');
+        }
+        const mediaDialog = new weWidgets.MediaDialog(this, {
+            noIcons: true,
+            noDocuments: true,
+            noVideos: !this.allowVideos,
+            isForBgVideo: true,
+        }, dummyEl).open();
+        mediaDialog.on('save', this, data => {
+            if (data.bgVideoSrc) {
+                this.setValue(data.bgVideoSrc);
+                this.isVideo = true;
+            } else {
+                // Accessing the value directly through img.src converts the url to absolute
+                this.setValue(dummyEl.getAttribute('src'));
+                this.isVideo = false;
+            }
+            this._notifyValueChange(false);
+            // Have to manually update UI because this is asynchronous
+            this.updateUI();
+        });
+    },
+    /**
+     * Called when the remove image button is clicked.
+     *
+     * @private
+     */
+    _onRemoveImage: function (ev) {
+        this.setValue(this.getDefaultValue());
```

Comment:
`this._value = '';` seems better (the widget holds no value, but will still returns the default one is asked for its value anyway).